src/Core/Translation/Provider/ThemeCatalogueLayersProvider.php

```diff
@@ -0,0 +1,236 @@
+     * @param ThemeExtractor $themeExtractor
+     * @param ThemeRepository $themeRepository
+     * @param Filesystem $filesystem
+     * @param string $themeResourcesDir
+     * @param string $themeName
+     */
+    public function __construct(
+        CatalogueLayersProviderInterface $coreFrontProvider,
+        DatabaseTranslationLoader $databaseTranslationLoader,
+        ThemeExtractor $themeExtractor,
+        ThemeRepository $themeRepository,
+        Filesystem $filesystem,
+        string $themeResourcesDir,
+        string $themeName
+    ) {
+        $this->databaseTranslationLoader = $databaseTranslationLoader;
+        $this->coreFrontProvider = $coreFrontProvider;
+        $this->themeExtractor = $themeExtractor;
+        $this->themeRepository = $themeRepository;
+        $this->filesystem = $filesystem;
+        $this->themeResourcesDir = $themeResourcesDir;
+        $this->themeName = $themeName;
+
+        $this->assertThemeIsValid();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultCatalogue(
+        string $locale,
+        bool $refreshCache = false
+    ): MessageCatalogue {
+        // Extracts wordings from the theme's templates
+        $catalogue = $this->themeExtractor->extract($this->theme, $locale, $refreshCache);
+
+        return $this->normalize($catalogue);
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return MessageCatalogue
+     *
+     * @throws FileNotFoundException
+     */
+    public function getFileTranslatedCatalogue(string $locale): MessageCatalogue
+    {
+        // load front office catalogue
+        $catalogue = $this->coreFrontProvider->getFileTranslatedCatalogue($locale);
+
+        try {
+            $fileTranslatedCatalogue = (new FileTranslatedCatalogueProvider(
+                $this->getResourceDirectory(),
+                ['*']
+            ))
+                ->getCatalogue($locale);
+
+            // overwrite with the theme's own catalogue
+            $catalogue->addCatalogue($fileTranslatedCatalogue);
+        } catch (FileNotFoundException $e) {
+            // there are no translation files, ignore them
+            return new MessageCatalogue($locale);
+        }
+
+        return $catalogue;
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return MessageCatalogue
+     */
+    public function getUserTranslatedCatalogue(string $locale): MessageCatalogue
+    {
+        return (new UserTranslatedCatalogueProvider(
+            $this->databaseTranslationLoader,
+            ['*'],
+            $this->themeName
+        ))
+            ->getCatalogue($locale);
+    }
+
+    /**
+     * Check if theme is registered in DB and set class property
+     */
+    private function assertThemeIsValid(): void
+    {
+        try {
+            $theme = $this->themeRepository->getInstanceByName($this->themeName);
+            if (!$theme instanceof Theme) {
+                throw new PrestaShopException();
```

Comment:
Open question, shouldn't another exception?

Reply:
I approve this one.

Reply:
Ping @sowbiba